Ginger/Ginger/DataSource/ImportExcelWizardLib/ImportDataSourceSheetSelection.xaml.cs

```diff
@@ -72,36 +72,103 @@ public void WizardEvent(WizardEventArgs WizardEventArgs)
                         List<string> SheetsList = impParams.GetSheets(false);
                         SheetsList.Insert(0, "-- All --");
                         GingerCore.General.FillComboFromList(xSheetNameComboBox, SheetsList);
-                        if(SheetsList.Contains(mWizard.SheetName))
+                        if (SheetsList.Contains(mWizard.SheetName))
                         {
                             xSheetNameComboBox.SelectedIndex = SheetsList.IndexOf(mWizard.SheetName);
                         }
-                    }                    
+                    }
+                    break;
+                case EventType.LeavingForNextPage:
+                    mWizard = (ImportDataSourceFromExcelWizard)WizardEventArgs.Wizard;
+                    string excelPathForSpaceCheck = mWizard.Path;
+
+                    string sheetName = mWizard.SheetName;
+                    bool headingRow = mWizard.HeadingRow;
+                    bool isModelParamsFile = mWizard.IsModelParamsFile;
+                    bool isImportEmptyColumns = mWizard.IsImportEmptyColumns;
+                    WizardEventArgs.CancelEvent = CheckIfCancelEvent(excelPathForSpaceCheck, sheetName, headingRow, isImportEmptyColumns, isModelParamsFile);
                     break;
                 default:
                     break;
             }
         }
 
+        private bool CheckIfCancelEvent(string Path, string SheetName, bool HeadingRow, bool IsImportEmptyColumns, bool IsModelParamsFile)
+        {
+
+            ImportOptionalValuesForParameters impParams = new ImportOptionalValuesForParameters();
+            string outErrCol = string.Empty;
+            DataSet ExcelImportData;
+            impParams.ExcelFileName = Path;
+            if (SheetName != "-- All --")
+            {
+                impParams.ExcelSheetName = SheetName;
+                impParams.ExcelWhereCondition = String.Empty;
+                ExcelImportData = impParams.GetExcelAllSheetData(SheetName, HeadingRow, IsImportEmptyColumns, IsModelParamsFile);
+                if (ExcelImportData != null && ExcelImportData.Tables.Count >= 1)
+                {
+                    if (ExcelImportData.Tables.Count == 1)
```

Comment:
same condition is handled in if on line 108, please remove the duplicate condition